src/Adapter/Http/Repository/HttpPodRepository.php

```diff
@@ -176,12 +189,13 @@ public function exists($name)
      */
     public function attach(Pod $pod, callable $callable)
     {
-        while ($this->isPending($pod)) {
+        $this->statusProvider->setPod($pod);
```

Comment:
either this `$pod` should be in the constructor of your status provide or as an argument of its `isPending` method.